edx-solutions/xblock-drag-and-drop-v2
# Click handlers and backend processing for assessment mode

**Description:** This PR actually implements submitting attempts for Assessment Mode. As of this PR, the minimum viable product implementation of Assessment Mode in Drag and Drop v2 is complete.

**Dependencies:** #86 (now merged)

**JIRA:** Part of [SOL-1944](https://openedx.atlassian.net/browse/SOL-1944)

**Discussions**: [Discovery document for DnDv2 assessment mode](https://docs.google.com/document/d/1rFliZo1vlohJ7cde7KfP5bgEk0QqcvT1VNnUUa9_vjE/edit) (relevant section: 2.2.3)

**Sandbox URLs**: 
* LMS: http://dndv2-sandbox8.opencraft.hosting/
* Studio: http://studio-dndv2-sandbox8.opencraft.hosting/

Updated 2016-08-12 to commit 5b837e7

**Testing instructions:**
1. Create/edit DnDv2; put it into assessment mode
2. Set "Max attempts" to N: N > 2
3. Publish, open in LMS.
4. Note that "Submit" button is displayed. Make sure problem is in its initial state (reset if needed). Submit button should be disabled. Attempts info should not be displayed.
5. Put any item to correct zone. "Submit" button should become enabled.
6. Click "Submit". Expected to happen:
6.1. Submit button shows spinner until AJAX request is sent
6.2. When AJAX is completed, incorrect items are returned to item bank, other items stay where they were.
6.3. Feedback message is updated to contain info about correctly placed items, misplaced items and items that should be put, but was not. If there were any misplaced items, feedback should also contain note about them being returned to item bank. If "initial" overall feedback is specified, it should be shown as well.
6.4. Attempts info should increase the number of used attempts.
7. Use N-1 attempts to test various combinations of correct, misplaced and missing items (note pluralization should be appropriate). Same things as in (6) should happen.
8. Submit final attempt. Expected to happen:
8.1. Same as 6.1. - spinner
8.2. All items stay where they were (misplaced are not returned to bank, as opposed to 6.2)
8.3. All of 6.3 + note about using final attempt and final grade are shown.
8.4. Attempts info should read "Used N of N attempts".
8.5. Submit and Reset buttons are permanently disabled.

**Reviewers**
